src/commons/utils/alerts.py

```diff
@@ -0,0 +1,168 @@
+        body = f"""<h3>TestPlan {self.tp_id} {execution_status} on {self.health_obj.host}.</h2>\n
+                   <h3>PFA hctl cluster status, pod status & execution status.</h3>\n"""
+        build_url = os.getenv("BUILD_URL")
+        if build_url:
+            body += f"""Visit Jenkins Job: <a href="{build_url}">{build_url}</a>"""
+        message = MIMEMultipart()
+        message['From'] = self.sender
+        message['To'] = COMMASPACE.join(self.receiver.split(','))
+        message['Date'] = formatdate(localtime=True)
+        message['Subject'] = subject
+        if not self.message_id:
+            self.message_id = make_msgid()
+            message["Message-ID"] = self.message_id
+        else:
+            message["In-Reply-To"] = self.message_id
+            message["References"] = self.message_id
+
+        attachment = MIMEApplication(hctl_status, Name="hctl_status.txt")
+        attachment['Content-Disposition'] = 'attachment; filename=hctl_status.txt'
+        message.attach(attachment)
+
+        if result:
+            attachment = MIMEApplication(pod_status, Name="pod_status.txt")
+            attachment['Content-Disposition'] = 'attachment; filename=pod_status.txt'
+            message.attach(attachment)
+        else:
+            body += """<h3>Could not collect pod status</h3>"""
+        name = os.path.basename(self.report_path)
+        if os.path.exists(self.report_path):
+            message.attach(MIMEText(body, "html"))
+            with open(self.report_path, "rb") as fil:
+                attachment = MIMEApplication(fil.read(), Name=name)
+            attachment['Content-Disposition'] = f'attachment; filename={name}'
+            message.attach(attachment)
+        else:
+            message.attach(MIMEText(body + f"<h3>Could not find {self.report_path}.</h3>", "html"))
+        return message
+
+    def run(self):
+        """
+        Send Mail notification periodically.
+        """
+        while not self.event_fail.is_set():
+            message = self.prepare_email(execution_status="in progress")
+            self.mail_obj.send_mail(message)
+            current_time = time.time()
+            while time.time() < current_time + CORIO_CFG.email_interval_mins * 60:
+                if self.event_fail.is_set():
+                    break
+                time.sleep(60)
+        message = self.prepare_email(execution_status="failed")
+        self.mail_obj.send_mail(message)
+
+
+def send_mail_notification(sender_email, receiver_email, tp_id, corio_start_time):
+    """
+    Send mail notification
+    :param sender_email: Sender Mail ID
+    :param receiver_email: Receiver Mail ID
+    :param tp_id: Test Plan ID
+    :param corio_start_time:
+    :return MailNotification Process.
+    """
+    report_path = get_report_file_path(corio_start_time)
+    mail_notify = MailNotification(sender=sender_email, receiver=receiver_email, tp_id=tp_id,
```

Comment:
can't we keep test plan id as an optional parameter?

Reply:
It doesn't matter if the test plan ID is passed or not passed.
If not passed, we won't get it in the email. If passed, we will get it in the email.